java-frontend/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -705,26 +705,29 @@ private void executeUnaryExpression(Tree tree) {
     programState = unstackUnary.state;
     SymbolicValue unarySymbolicValue = constraintManager.createSymbolicValue(tree);
     unarySymbolicValue.computedFrom(unstackUnary.values);
-    if (tree.is(Tree.Kind.POSTFIX_DECREMENT, Tree.Kind.POSTFIX_INCREMENT, Tree.Kind.PREFIX_DECREMENT, Tree.Kind.PREFIX_INCREMENT)
-      && ((UnaryExpressionTree) tree).expression().is(Tree.Kind.IDENTIFIER)) {
-      programState = programState.put(((IdentifierTree) ((UnaryExpressionTree) tree).expression()).symbol(), unarySymbolicValue);
-    }
     if (tree.is(Tree.Kind.POSTFIX_DECREMENT, Tree.Kind.POSTFIX_INCREMENT)) {
       programState = programState.stackValue(unstackUnary.values.get(0));
     } else {
       programState = programState.stackValue(unarySymbolicValue);
     }
+    if (tree.is(Tree.Kind.POSTFIX_DECREMENT, Tree.Kind.POSTFIX_INCREMENT, Tree.Kind.PREFIX_DECREMENT, Tree.Kind.PREFIX_INCREMENT)
+      && ((UnaryExpressionTree) tree).expression().is(Tree.Kind.IDENTIFIER)) {
+      programState = programState.put(((IdentifierTree) ((UnaryExpressionTree) tree).expression()).symbol(), unarySymbolicValue);
+    }
   }
 
   private void executeIdentifier(IdentifierTree tree) {
     Symbol symbol = tree.symbol();
     SymbolicValue value = programState.getValue(symbol);
     if (value == null) {
       value = constraintManager.createSymbolicValue(tree);
-      programState = programState.put(symbol, value);
+      programState = programState.stackValue(value);
       learnIdentifierNullConstraints(tree, value);
+      programState = programState.put(symbol, value);
+    } else {
+      programState = programState.stackValue(value);
+      programState = programState.put(symbol, value);
```

Comment:
The call to `put` can be extracted from the `if-else` statement in order to remove the duplication.